capstone/src/main/java/com/google/step/servlets/RestaurantSignUpServlet.java

```diff
@@ -0,0 +1,93 @@
+package com.google.step.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.GeoPt;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Servlet that will put the restaurant information in Datastore.
+ */
+@WebServlet("/restaurantSignUp")
+public class RestaurantSignUpServlet extends HttpServlet {
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+
+    // Only available if the user is logged in.
+    if (!userService.isUserLoggedIn()) {
+      response.setStatus(HttpServletResponse.SC_FORBIDDEN);
+      return;
+    }
+
+    String id = userService.getCurrentUser().getUserId();
+    String email = userService.getCurrentUser().getEmail();
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    // TODO: add a check if the email already exists as a user
+    // If the restaurant email already exists in datastore don't add it.
+    Query query =
+        new Query("RestaurantUser")
+            .setFilter(new Query.FilterPredicate("email", Query.FilterOperator.EQUAL, email));
+    PreparedQuery results = datastore.prepare(query);
+    Entity resultEntity = results.asSingleEntity();
+    if (resultEntity != null) {
+      response.setStatus(HttpServletResponse.SC_NOT_ACCEPTABLE);
```

Comment:
If not found, should we be using SC_NOT_FOUND instead?

Reply:
Yeah you're right.

Reply:
This was also taken out since we won't be needing that check.